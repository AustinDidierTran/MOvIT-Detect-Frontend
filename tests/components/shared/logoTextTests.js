/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import React from 'react';
import PropTypes from 'prop-types';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import LogoText from '../../../src/components/shared/logoText';

Enzyme.configure({ adapter: new Adapter() });

describe('LogoText Tests', () => {
  const spy = sinon.spy();
  const props = {
    value: 'test',
    iconClass: 'fa fa-id-card',
    placeHolder: 'test',
    onChange: (value) => { spy(value); },
  };

  it('should match the snapshot', () => {
    const wrapper = shallow(<LogoText {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have proptypes', () => {
    const actualValue = LogoText.propTypes;

    const expectedValue = {
      placeHolder: PropTypes.string.isRequired,
      iconClass: PropTypes.string.isRequired,
      value: PropTypes.string,
      onChange: PropTypes.func.isRequired,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('input prints the right value', () => {
    const wrapper = shallow(<LogoText {...props} />);

    expect(wrapper.find('#logoText').props().value).toEqual('test');
  });

  it('input calls the method onChange with the right value', () => {
    const wrapper = shallow(<LogoText {...props} />);

    wrapper.find('#logoText').simulate('change', { target: { value: 'testing' } });

    expect(spy.calledOnce).toEqual(true);
    expect(spy.getCalls()[0].args[0]).toEqual('testing');
  });
});
