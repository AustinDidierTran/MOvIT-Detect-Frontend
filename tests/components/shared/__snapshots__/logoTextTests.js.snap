// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoText Tests should match the snapshot 1`] = `
<div
  className="form-horizontal row mb-3"
>
  <div
    className="col-1 d-inline-block text-right"
  >
    <span
      style={
        Object {
          "fontSize": "large",
          "minWidth": "26px",
          "paddingTop": "6px",
        }
      }
    >
      <i
        className="fa fa-id-card"
      />
    </span>
  </div>
  <div
    className=" col-11 col-lg-8 d-inline-block pr-1"
  >
    <input
      className="form-control"
      id="logoText"
      onChange={[Function]}
      placeholder="test"
      type="text"
      value="test"
    />
  </div>
</div>
`;
